File: multi-module/api/src/main/java/com/module/member/MemberApi.java
package com.module.member;

import com.module.core.member.Member;
import com.module.core.member.MemberRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/members")
@RequiredArgsConstructor
public class MemberApi {


  private final MemberRepository memberRepository;

  @PostMapping
  public Member create() {
    final Member member = new Member("test");
    return memberRepository.save(member);
  }


}

File: spring-test/src/main/kotlin/com/spring/test/member/MemberApi.kt
package com.spring.test.member

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/members")
class MemberApi(
    private val memberRepository: MemberRepository,
    private val memberService: MemberService
) {

    @GetMapping
    fun getMembers(pageable: Pageable): Page<Member> {
        return memberRepository.findAll(pageable)
    }

    @PostMapping
    fun createMember(@RequestBody dto: MemberRequest) {
        memberService.create(dto.name, dto.email)
    }
}

data class MemberRequest(
    val name: String,
    val email: String
)